source/features/forked-to.tsx

```diff
@@ -30,8 +32,19 @@ const updateCache = cache.function(async (): Promise<string[] | undefined> => {
 	staleWhileRevalidate: 5
 });
 
+async function updateForkLinkToDefaultBranch(url: GitHubURL): Promise<void> {
+	if (await doesFileExist(url)) {
+		const defaultBranch = await getDefaultBranch(getRepositoryInfo(url.pathname.slice(1)));
+
+		select<HTMLAnchorElement>(`.pagehead-actions [href*="${url.pathname}"]`)!.pathname = url.assign({
+			route: 'edit',
+			branch: defaultBranch
+		}).pathname;
```

Comment:
`createLink` is called multiple times in a `map` below, which means that there are multiple links. Use `select.all`

Reply:
The way it's written now it gets called for each link separately.

Reply:
You're right. However this needs to be made more readable though:
```suggestion
		url.assign({
			branch: defaultBranch
		});
		select<HTMLAnchorElement>(`.pagehead-actions [href*="${url.pathname}"]`)!.pathname = url.pathname;
```

But really this way of changing the URL is complete spaghetti. `createLink` is a getter, it shouldn't also trigger an HTTP request and query the DOM independently. This also applies to the previous similar PR, but it made it a little awkward there with the JSX.

A more reasonable way would be something like:

```js
const link = (
	<a
		className="btn btn-sm float-left rgh-forked-button"
		title={`Open your fork at ${forks[0]}`}
	>
		<LinkExternalIcon/>
	</a>
);
forkCounter.before(link);
setUrl(link);
```

`setUrl` would then first set the `HEAD` URL and then update it if necessary.

Reply:
If you have any p/react experience this might be a good place to use Preact instead of `dom-chef`, it would _probably_ make these kind of updates more sensible.

Reply:
I have none

Reply:
Then the `setUrl` suggestion still applies